tests/integration.rs

```diff
@@ -2337,6 +2337,63 @@ mod tests {
             handle_child_output(r, &output);
         }
 
+        #[test]
+        #[cfg(target_arch = "aarch64")]
+        fn test_aarch64_guest_numa_nodes_dt() {
```

Comment:
Why do you need a dedicated test?
You should be able to reuse `test_guest_numa_nodes()`.
I think you should reuse `test_guest_numa_nodes()` to make it a common function that is called from `test_guest_numa_nodes_acpi()` and `test_guest_numa_nodes_dt()`. Of course you would apply the correct feature gating on top of each to make sure they can run as expected.

Reply:
Ah, this is because we want to make sure that both the DT and ACPI will be tested on Arm. For now, we only build the Cloud Hypervisor for once in CI using `cargo build`. Using feature gates cannot cover both cases.

But I like the idea of reusing `test_guest_numa_nodes()` as a common function. Let me do some refactoring :)) Thank you very much for the suggestion!

Reply:
Can you can test the two code paths by firmware boot vs direct boot (acpi vs fdt)?

Reply:
> Can you can test the two code paths by firmware boot vs direct boot (acpi vs fdt)?

Hi @rbradford ! Not very sure what you mean (as there is a typo), but yeah, the idea is just to change the path of `--kernel`. However on arm we need to run both of two cases, i.e. `--kernel=direct_kernel_boot_path()` once and `--kernel=edk2_path()` once. I don't think this is possible in one single test case unless using a loop. So I would extract most of the common code into a helper function and split the original test case into `test_guest_numa_nodes_acpi()` and `test_guest_numa_nodes_dt()`

Reply:
This is exactly what I was thinking.